### ISSUE TITLE
"-noNotifyFalseUpdate" - notification cycles a certain number of times

### ISSUE THREAD

**Issue by sekiguchi-yohei** [Fiware/context.Orion-LD#917]
Hi.

An unintended event occurred with "-noNotifyFalseUpdate".
If the value field of an attribute of an entity sent in a notification is an object, the notification was repeated a certain number of times.
(Same event as in issues/907)
https://github.com/FIWARE/context.Orion-LD/issues/907

NG
```
{
    "id": "urn:ngsi-ld:Person:person001",
    "type": "Person",
    "name": {
        "type": "Property",
        "value": "person001"
    },
    "birthDate": {
        "type": "Property",
        "value": {
            "type": "DateTime",
            "value": "1992-07-16T11:00:00.00Z"
        }
    }
}
```

version
```
{
  "orionld version": "post-v0.8.1",
  "orion version":   "1.15.0-next",
  "uptime":          "0 d, 18 h, 45 m, 18 s",
  "git_hash":        "nogitversion",
  "compile_time":    "Mon Jul 26 20:31:37 UTC 2021",
  "compiled_by":     "root",
  "compiled_in":     "buildkitsandbox",
  "release_date":    "Mon Jul 26 20:31:37 UTC 2021",
  "doc":             "https://fiware-orion.readthedocs.org/en/master/"
}
```

**Comment by kzangeli** [comment]
Yeah, that is what is implemented.
I implemented for Orion-LD exactly was is implemented for Orion and that is a comparison old/new value if not compound.

It's a lot more effort to compare compound values. But, if this is what you need, I'll have to implement it.

**Comment by kzangeli** [comment]
Yeah, that is what is implemented.
I implemented for Orion-LD exactly was is implemented for Orion and that is a comparison old/new value if not compound.

It's a lot more effort to compare compound values. But, if this is what you need, I'll have to implement it.

**Comment by sekiguchi-yohei** [comment]
Hi. 

Sorry
I hadn't done a similar test with V2...（value field of an attribute of an entity）
It will be difficult for you to implement this, so we will look into other ways to achieve this.

Thank you very much.

**Comment by kzangeli** [comment]
Yeah, it is far from trivial. A recursive tree comparison algorithm on two trees that must first be sorted ... Only, how do you sort a possible mix of strings, numbers, bools, objects, arrays ...

However, it's doable and not a bad addition to the broker, so, let's keep this issue open and once I'm back from my parental leave (had a daughter, Emily, yesterday :)) I will try to find time to implement this

**Comment by sekiguchi-yohei** [comment]
Congratulations on your new baby!
I'm really looking forward to seeing it implemented.

**Comment by kzangeli** [comment]
So, I have the implementation ready and working in my developer machine.
For some reason the functest for this "new feature" fails in github actions.
With some luck I'll find the reason for this soon and can merge the pull request into develop!
It depends a little ( a lot! ) on how much time my wife lets me have behind the keyboard ... :)

**Comment by sekiguchi-yohei** [comment]
Thanks for the fix!
Looking forward to your revised version!
（However, please don't try too hard and upset your wife...）

**Comment by kzangeli** [comment]
So, hopefully fixed.
Bear in mind that arrays cannot be sorted, i.e. `[ 1, 2 ]` is not the same as `[ 2, 1 ]`.
Objects are sorted.
Please test and let me know (by a comment and closing of the issue if all is good).

**Comment by sekiguchi-yohei** [comment]
Thanks for the quick response!
After testing, I can confirm that the problem has been resolved.
